gwpy/types/tests/test_series.py

```diff
@@ -238,6 +238,41 @@ def test_crop_float_precision(self):
         cropped = series.crop(end=1.)
         utils.assert_quantity_equal(series[:-1], cropped)
 
+        # test that floating point division followed by floor doesn't result
+        # in cropping one too many indices
+        arrlen = 500
+        xmax = 0.508463154883984
+        x_series = numpy.linspace(0, xmax, arrlen)
+        series = Series([0] * arrlen, xindex=x_series)
+        expected = series.xindex[-2]
+        assert series.crop(end=xmax).xindex[-1] == expected
+
+        # test that cropping at a point between two grid values is floored
+        # e.g. x = [1, 2, 3], end = 2.5, result = [1, 2]
+        series = Series([0] * 3, xindex=[1, 2, 3])
+        print(series.crop(end=2.5))
+        assert all(series.crop(end=2.5).xindex == [1, 2])
+
+        series = Series([0] * 3, xindex=[1, 2, 3])
+        assert all(series.crop(start=2.5).xindex == [3])
+
+        series = Series([0] * 5, xindex=[1, 2, 3, 4, 5])
+        assert all(series.crop(start=2.5, end=4.5).xindex == [3, 4])
+
+        # some slightly harder examples
+        xmax = 0.508463154883984
+        x_series = numpy.linspace(0, xmax, arrlen)
+        series = Series([0] * arrlen, xindex=x_series)
+        expected = series.xindex[-2]
+        mid = (0.6 * series.xindex[-2] + 0.4 * series.xindex[-1])
+        assert series.crop(end=mid).xindex[-1] == expected
+
+        xmin = 0.508463154883984
+        x_series = numpy.linspace(xmin, 1.0, arrlen)
+        series = Series([0] * arrlen, xindex=x_series)
+        expected = series.xindex[0]
+        assert series.crop(start=xmin).xindex[0] == expected
```

Comment:
Can you please separate these new tests into at least one new test method, so that they can be evaluated separately to the existing `crop` test(s)?